src/mbedtls_stream.c

```diff
@@ -0,0 +1,474 @@
+/*
+ * Copyright (C) the libgit2 contributors. All rights reserved.
+ *
+ * This file is part of libgit2, distributed under the GNU GPL v2 with
+ * a Linking Exception. For full terms see the included COPYING file.
+ */
+
+#ifdef GIT_MBEDTLS
+
+#include <ctype.h>
+
+#include "global.h"
+#include "stream.h"
+#include "socket_stream.h"
+#include "git2/transport.h"
+
+#ifdef GIT_CURL
+# include "curl_stream.h"
+#endif
+
+#include "mbedtls/config.h"
+#include <mbedtls/x509.h>
+#include <mbedtls/x509_crt.h>
+#include <mbedtls/error.h>
+#include "mbedtls/net.h"
+#include "mbedtls/debug.h"
+#include "mbedtls/entropy.h"
+#include "mbedtls/ctr_drbg.h"
+#include "mbedtls/error.h"
+#include "mbedtls/certs.h"
+
+#ifndef OPENSSLDIR
+# define OPENSSLDIR              "/usr/lib/ssl"
```

Comment:
cross-linking to https://github.com/JuliaLang/julia/issues/18693, this isn't very uniform across distributions and we might want to come up with a better certificate search mechanism?